privacy/ccpa/policy.go

```diff
@@ -21,17 +39,41 @@ func ReadPolicy(req *openrtb.BidRequest) (Policy, error) {
 	if req != nil && req.Regs != nil && len(req.Regs.Ext) > 0 {
 		var ext openrtb_ext.ExtRegs
 		if err := json.Unmarshal(req.Regs.Ext, &ext); err != nil {
-			return policy, err
+			return Policy{}, err
```

Comment:
My intention was to keep uniformity with the newly added second loop. I didn't want the second error to result in a partial policy. I can tweak the code to avoid the duplicate allocation.. but this also brings me to rethink error handling.

If the NoSale section is malformed, PBS should still enforce the CCPA signal it received. I'll change the flow here.

Reply:
Fixed.

Reply:
I also changed the code to ignore NoSale errors. It's very unlikely (impossible?) to happen with the request validation in the auction endpoints. I considered adding a warning err, but there's adapters using this method now and getting an error about NoSale extensions when they mostly just care about the `Value` would be evil.

Reply:
I think we should also discuss json marshaling at a larger scale. There's a lot of duplicated effort with this throughout the code.